src/scripts/main.js

```diff
@@ -1,11 +1,22 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener(eventName, (e) => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
+  });
 }
```

Comment:
The waitFor function only resolves once, after the first event occurrence. If the event occurs multiple times, the function will not resolve again. This can be problematic if you want to track an event multiple times. Consider modifying this function to continuously listen for the event and resolve every time it occurs.